### ISSUE TITLE
MPI_Win_create fails with MPI_ERR_WIN for v4.1.0 and ob1

### ISSUE THREAD

**Issue by PavelStishenko** [open-mpi/ompi#11466]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

 v4.1.0

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

OpenMPI installed from Debian 11 (bullseye) repo

### Please describe the system on which you are running

* Operating system/version: Debian 11 (bullseye) 
* Computer hardware: ThinkStation C20x
* Network type: WiFi via USB dongle

-----------------------------

## Details of the problem

The following code (from issue #9871):
```c

#include <mpi.h>

int main(int argc, char *argv[]) {
  void   *buf;
  MPI_Win window;

  MPI_Init(&argc, &argv);
  MPI_Alloc_mem(16, MPI_INFO_NULL, &buf);
  MPI_Win_create(buf, 16, 1, MPI_INFO_NULL, MPI_COMM_WORLD, &window);
  MPI_Win_free(&window);
  MPI_Free_mem(buf);
  MPI_Finalize();
}
```
compiled as 
```
gcc testwin2.c  `pkg-config --cflags --libs ompi-c`   -o testwin.x
```
being launched as 
```
mpiexec.openmpi -n 2 --mca osc_base_verbose 100 --mca pml_base_verbose 100  --mca btl_base_verbose 100  ./testwin.x
```

fails with error messages:
```
[localhost:43128] *** An error occurred in MPI_Win_create
[localhost:43128] *** reported by process [2029780993,1]
[localhost:43128] *** on communicator MPI_COMM_WORLD
[localhost:43128] *** MPI_ERR_WIN: invalid window
[localhost:43128] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[localhost:43128] ***    and potentially your MPI job)
```

Full error  log and the output of `ompi_info` are attached:
[testwin.log](https://github.com/open-mpi/ompi/files/10885323/testwin.log)

[ompi_info.log](https://github.com/open-mpi/ompi/files/10885326/ompi_info.log)

**Comment by devreal** [comment]
@PavelStishenko Thanks for the report! Can you please try the latest release in the 4.1.x series? 4.1.5 was just released: https://www.open-mpi.org/software/ompi/v4.1/

Also, I don't think ob1 has anything to do with that (it's the point to point module, not RMA). Are you building against UCX?

**Comment by PavelStishenko** [comment]
@devreal thanks for the response!
I am not sure that I am capable to build OpenMPI from source. I used `apt install` to install it. Also I have the same OpenMPI version on another PC with Fedora, where is works fine.  The only difference is that Fedora uses `ucx`, and Debian uses `ob1`. 

Here are the same logs from Fedora:
[testwin.log](https://github.com/open-mpi/ompi/files/10885476/testwin.log)
[ompi_info.log](https://github.com/open-mpi/ompi/files/10885475/ompi_info.log)

**Comment by PavelStishenko** [comment]
OK, it works for OpenMPI-4.1.5. So the bug probably can be closed, but it is interesting why 4.1.0 works for Fedora with both PMLs `ob1`, and `ucx` and none of them works with Debian. Sounds like a bug for Debian.

**Comment by devreal** [comment]
Thanks for checking 4.1.5. Open MPI 4.1.0 is over 2 years old, that's a long time for software like Open MPI (i.e., many bugs have been fixed in the meantime).
